pgmpy/estimators/HillClimbSearch.py

```diff
@@ -283,24 +272,33 @@ def estimate(
                 "'start_dag' should be a DAG with the same variables as the data set, or 'None'."
             )
 
-        # Step 1.3: Check fixed_edges
-        if not hasattr(fixed_edges, "__iter__"):
-            raise ValueError("fixed_edges must be an iterable")
+        if expert_knowledge:
+            # Step 1.3: Check fixed_edges
+            if expert_knowledge.fixed_edges == set():
+                fixed_edges = set()
+            else:
+                fixed_edges = set(expert_knowledge.fixed_edges)
+                start_dag.add_edges_from(fixed_edges)
+                if not nx.is_directed_acyclic_graph(start_dag):
+                    raise ValueError(
+                        "fixed_edges creates a cycle in start_dag. Please modify either fixed_edges or start_dag."
+                    )
+
+            # Step 1.4: Check black list and white list
+            black_list = (
+                set()
+                if expert_knowledge.black_list is None
+                else set(expert_knowledge.black_list)
+            )
+            white_list = (
+                set([(u, v) for u in self.variables for v in self.variables])
+                if expert_knowledge.white_list is None
+                else set(expert_knowledge.white_list)
+            )
         else:
-            fixed_edges = set(fixed_edges)
-            start_dag.add_edges_from(fixed_edges)
-            if not nx.is_directed_acyclic_graph(start_dag):
-                raise ValueError(
-                    "fixed_edges creates a cycle in start_dag. Please modify either fixed_edges or start_dag."
-                )
-
-        # Step 1.4: Check black list and white list
-        black_list = set() if black_list is None else set(black_list)
-        white_list = (
-            set([(u, v) for u in self.variables for v in self.variables])
-            if white_list is None
-            else set(white_list)
-        )
+            white_list = set([(u, v) for u in self.variables for v in self.variables])
```

Comment:
I would suggest avoiding this and leave it as None. Because we will have to check whether an edge is in the white_list set in every iteration of the algorithm, it can be computationally expensive. It would be faster to just check if the value is None and only check if the edge is in white_list if it is not None.

Reply:
I guess for this I will have to slightly change the [legal operation calculation step](https://github.com/pgmpy/pgmpy/blob/cf8d0f12e2e5be62b01ff8fded85f3f64eab1e84/pgmpy/estimators/HillClimbSearch.py#L61). I'll make the change.

Reply:
Will make the changes today!